addons/account/models/account_payment.py

```diff
@@ -739,32 +751,25 @@ def build_query(move_table_and_alias, outstanding_account_ids, payments):
                    AND move_line.partner_id = dup_move_line.partner_id
                    AND move_line.company_id = dup_move_line.company_id
                    AND move_line.date = dup_move_line.date
-                   AND dup_move_line.parent_state IN %(matching_states)s
+                   AND dup_move_line.parent_state = ANY(%(matching_states)s)
                    AND (
                        move_line.account_id = dup_move_line.account_id
-                       OR dup_move_line.account_id = %(suspense_account_id)s
-                       OR dup_move_line.account_id IN %(outstanding_account_ids)s
+                       OR dup_move_line.account_id = ANY(%(additional_account_ids)s)
                    )
                    AND NOT dup_move_line.reconciled
-                 WHERE move_line.payment_id IN %(payments)s
-                   AND (
-                       -- Case 1: a move is a credit in same account receivable or debit in same acc payable as the payment
-                       (dup_move_line.account_id NOT IN %(outstanding_account_ids)s AND move_line.balance = dup_move_line.balance)
-                       OR
-                       -- Case 2: a move is a credit in outstanding receipts or debit in outstanding payments
-                       (dup_move_line.account_id IN %(outstanding_account_ids)s AND move_line.balance = -1.0 * dup_move_line.balance)
-                   )
+                 WHERE move_line.payment_id = ANY(%(payments)s)
+                   %(extra_conditions)s
                    AND (
                        move_line.payment_type = 'inbound' AND dup_move_line.balance < 0.0
                        OR move_line.payment_type = 'outbound' AND dup_move_line.balance > 0.0
                    )
               GROUP BY move_line.payment_id
             """,
                 move_table_and_alias=move_table_and_alias,
-                matching_states=tuple(matching_states),
-                suspense_account_id=suspense_account_id,
-                outstanding_account_ids=outstanding_account_ids,
-                payments=tuple(payments),
+                matching_states=list(matching_states),
+                additional_account_ids=suspense_account_ids + outstanding_account_ids,
```

Comment:
```suggestion
                account_ids=suspense_account_ids + outstanding_account_ids,
```
nitpick: feel free to ignore